test/micro/org/openjdk/bench/java/lang/invoke/MethodHandleProxiesAsIFInstanceCreate.java

```diff
@@ -33,84 +33,63 @@
+import static java.lang.invoke.MethodType.methodType;
+
 /**
- * Benchmark evaluates the performance of MethodHandleProxies.*
+ * Benchmark evaluates the performance of MethodHandleProxies.asInterfaceInstance creation performance
  */
 @BenchmarkMode(Mode.AverageTime)
 @OutputTimeUnit(TimeUnit.NANOSECONDS)
 @State(Scope.Thread)
 @Warmup(iterations = 10, time = 1, timeUnit = TimeUnit.SECONDS)
 @Measurement(iterations = 5, time = 2, timeUnit = TimeUnit.SECONDS)
 @Fork(3)
-public class MethodHandleProxiesAsIFInstance {
-
+public class MethodHandleProxiesAsIFInstanceCreate {
     /**
-     * Implementation notes:
-     *   - asInterfaceInstance() can only target static MethodHandle (adapters needed to call instance method?)
-     *   - baselineCompute will quickly degrade to GC test, if escape analysis is unable to spare the allocation
-     *   - testCreate* will always be slower if allocation is not eliminated; baselineAllocCompute makes sure allocation is present
+     * Avoids elimination of computation
      */
-
     public int i;
 
+    private static final Lookup LOOKUP = lookup();
+    private static final MethodType MT_Doable = methodType(Doable.class);
+    private static final MethodType MT_int_int = methodType(int.class, int.class);
     private MethodHandle target;
-    private Doable precreated;
 
     @Setup
     public void setup() throws Throwable {
-        target = MethodHandles.lookup().findStatic(MethodHandleProxiesAsIFInstance.class, "doWork", MethodType.methodType(int.class, int.class));
-        precreated = MethodHandleProxies.asInterfaceInstance(Doable.class, target);
+        target = LOOKUP.findStatic(MethodHandleProxiesAsIFInstanceCreate.class, "doWork", MT_int_int);
     }
 
     @Benchmark
-    public Doable testCreate() {
-        Doable doable = MethodHandleProxies.asInterfaceInstance(Doable.class, target);
-        return doable;              // make sure allocation happens
+    public Doable createLambda() throws Throwable {
+        return (Doable) LambdaMetafactory.metafactory(LOOKUP, "doWork", MT_Doable, MT_int_int, target, MT_int_int).getTarget().invokeExact();
     }
 
     @Benchmark
-    public Doable testCreateCall() {
-        Doable doable = MethodHandleProxies.asInterfaceInstance(Doable.class, target);
-        i = doable.doWork(i);       // make sure computation happens
-        return null;                // let allocation be eliminated
+    public Doable createInterfaceInstance() {
+        return MethodHandleProxies.asInterfaceInstance(Doable.class, target);
     }
 
     @Benchmark
-    public Doable testCall() {
-        i = precreated.doWork(i);   // make sure computation happens
-        return precreated;
+    public Doable createCallLambda() throws Throwable {
+        Doable doable = (Doable) LambdaMetafactory.metafactory(LOOKUP, "doWork", MT_Doable, MT_int_int, target, MT_int_int).getTarget().invokeExact();
+        i = doable.doWork(i);
+        return doable;
     }
```

Comment:
In this case, instead of writing to `i`, the benchmark method can accept a `Blackhole` argument and use that to consume the value.
```suggestion
    public Doable createCallLambda(Blackhole bh) throws Throwable {
        Doable doable = (Doable) LambdaMetafactory.metafactory(LOOKUP, "doWork", MT_Doable, MT_int_int, target, MT_int_int).getTarget().invokeExact();
        bh.consume(doable.doWork(i));
        return doable;
    }
```